components/scream/src/physics/shoc/atmosphere_macrophysics.cpp

```diff
@@ -112,21 +386,27 @@ void SHOCMacrophysics::set_required_field_impl (const Field<const Real>& f) {
   // at the beginning of the run call. Other than that, there would be really
   // no need to store a scream field here; we could simply set the view ptr
   // in the Homme's view, and be done with it.
-  m_shoc_fields_in.emplace(f.get_header().get_identifier().name(),f);
+  const auto& name = f.get_header().get_identifier().name();
+  m_shoc_fields_in.emplace(name,f);
+  m_shoc_host_views_in[name] = Kokkos::create_mirror_view(f.get_view());
+  m_raw_ptrs_in[name] = m_shoc_host_views_in[name].data();
```

Comment:
Does shoc still need raw ptrs? Besides not being necessary, they are getting more dangerous to use now. Tracers can be subviews of a big Q array, but the subview is not contiguous in memory (the layout is (icol,iq,ilev)), so the ptr alone is not enough to access the array at the right location.

I'm assuming ptrs were used for calling f90. If that's still necessary, we need to allow tracers to be allocated not contiguously, OR have shoc copy them manually in/out to/from a contiguous array before/after calling f90.

Reply:
You are correct, they are unnecessary. I'll remove.